packages/strapi-plugin-users-permissions/controllers/User.js

```diff
@@ -97,9 +97,9 @@ module.exports = {
   update: async (ctx, next) => {
     try {
       if (strapi.plugins['users-permissions'].config.advanced.unique_email && ctx.request.body.email) {
-        const user = await strapi.query('user', 'users-permissions').findOne({ email: ctx.request.body.email });
+        const users = await strapi.plugins['users-permissions'].services.user.fetchAll({ email: ctx.request.body.email });
 
-        if (user) {
+        if (users && _.find(users, user => (user.id || user._id).toString() !== ctx.params.id)) {
```

Comment:
`if (_.find(users, user => (user.id || user._id).toString() !== ctx.params.id))`

The first condition was useless because `users` is an array and whatever the value of the array it will always return `true`.

Reply:
Also, should we apply the toString() method to `ctx.params.id` for SQL databases with integer ID?